cmd/fastqFilter/fastqFilter.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/fastq"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/numbers"
+	"log"
+	"math/rand"
+	"sync"
+)
+
+type Settings struct {
+	InFile    string
+	OutFile   string
+	R1InFile  string
+	R2InFile  string
+	R1OutFile string
+	R2OutFile string
+	PairedEnd bool
+	SubSet    float64
+	RandSeed  bool
+	SetSeed   int64
+	MinSize   int
+	MaxSize   int
+}
+
+func fastqFilter(s Settings) {
+	common.RngSeed(s.RandSeed, s.SetSeed)
+	var r float64
+
+	if s.PairedEnd {
+		ReadCh := make(chan *fastq.PairedEnd, 100000)
+		WriteCh := make(chan *fastq.PairedEnd, 100000)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go fastq.PairEndToChan(s.R1InFile, s.R2InFile, ReadCh)
+		go fastq.WritingChan(s.R1OutFile, s.R2OutFile, WriteCh, &wg)
+		for i := range ReadCh {
+			r = rand.Float64()
```

Comment:
You could also make this a bit more efficient by moving r generation and r > s.Subset check below the other checks. That way you only spend time trying to subset valid fastqs rather than trying to subset all fastqs.